File: python/gigasecond/gigasecond.py
"""
 --- Instructions --- 

Your task is to determine the date and time one gigasecond after a certain date.
A gigasecond is one thousand million seconds. 
That is a one with nine zeros after it.

If you were born on January 24th, 2015 at 22:00 (10:00:00pm), 
then you would be a gigasecond old on October 2nd, 2046 at 23:46:40 (11:46:40pm).
"""


from datetime import timedelta

GIGASECOND = 1_000_000_000

def add(moment):

    """
        Returns a date with one Gigasecond added to it.
    """

    diff = timedelta(seconds=GIGASECOND)

    return moment + diff

File: java/gigasecond/src/main/java/Gigasecond.java
import java.time.temporal.ChronoUnit;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

public class Gigasecond {

    private LocalDateTime moment;

    public Gigasecond(LocalDate moment) {
        this(LocalDateTime.of(moment, LocalTime.of(0, 0, 0)));
    }

    public Gigasecond(LocalDateTime moment) {
        this.moment = moment;
    }

    public LocalDateTime getDateTime() {
        long gigasecond = 1_000_000_000;
        return this.moment.plus(gigasecond, ChronoUnit.SECONDS);
    }
}
